clarity/contracts/pool-token/ytp-yield-wbtc-59760-wbtc.clar

```diff
@@ -1,89 +1,78 @@
+
+(define-read-only (get-total-supply)
+  (ok (ft-get-supply ytp-yield-wbtc-59760-wbtc))
 )
 
-(define-read-only (get-symbol)
-  (ok "yield-wbtc-59760-wbtc")
+(define-read-only (get-name)
+  (ok "ytp-yield-wbtc-59760-wbtc")
 )
 
-(define-read-only (get-expiry)
-  (ok 5976000000000)
+(define-read-only (get-symbol)
+  (ok "ytp-yield-wbtc-59760-wbtc")
 )
 
 (define-read-only (get-decimals)
   (ok u6)
 )
 
-(define-read-only (get-balance (owner principal))
-  (ok (ft-get-balance yield-wbtc-59760-wbtc owner))
+(define-read-only (get-balance (account principal))
+  (ok (ft-get-balance ytp-yield-wbtc-59760-wbtc account))
 )
 
-(define-read-only (get-total-supply)
-  (ok (ft-get-supply yield-wbtc-59760-wbtc))
+(define-public (set-token-uri (value (string-utf8 256)))
+  (begin
+    (asserts! (is-eq tx-sender (var-get contract-owner)) ERR-NOT-AUTHORIZED)
+    (ok (var-set token-uri value))
+  )
 )
 
 (define-read-only (get-token-uri)
-  (ok (some u"https://docs.alexgo.io/"))
+  (ok (some (var-get token-uri)))
 )
 
-
-;; one stop function to gather all the data relevant to the LP token in one call
-(define-read-only (get-data (owner principal))
-  (ok {
-    name: (unwrap-panic (get-name)),
-    symbol: (unwrap-panic (get-symbol)),
-    decimals: (unwrap-panic (get-decimals)),
-    uri: (unwrap-panic (get-token-uri)),
-    supply: (unwrap-panic (get-total-supply)),
-    balance: (unwrap-panic (get-balance owner))
-  })
+(define-public (transfer (amount uint) (sender principal) (recipient principal) (memo (optional (buff 34))))
+  (begin
+    (asserts! (is-eq sender tx-sender) ERR-NOT-AUTHORIZED)
+    (match (ft-transfer? ytp-yield-wbtc-59760-wbtc amount sender recipient)
+      response (begin
+        (print memo)
+        (ok response)
+      )
+      error (err error)
+    )
+  )
 )
 
-;; the extra mint method used when adding liquidity
-;; can only be used by arkadiko swap main contract
 (define-public (mint (recipient principal) (amount uint))
   (begin
-    (print "usda-token-swap.mint")
-    (print contract-caller)
-    (print amount)
-    ;; TODO - make dynamic
-    ;; (asserts! (is-eq contract-caller .arkadiko-swap-v1-1) (err ERR-NOT-AUTHORIZED))
-    (ft-mint? yield-wbtc-59760-wbtc amount recipient)
+    (asserts! (is-eq contract-caller (var-get contract-owner)) ERR-NOT-AUTHORIZED)
+    (ft-mint? ytp-yield-wbtc-59760-wbtc amount recipient)
   )
 )
 
-
-;; the extra burn method used when removing liquidity
-;; can only be used by arkadiko swap main contract
-(define-public (burn (recipient principal) (amount uint))
+(define-public (burn (sender principal) (amount uint))
   (begin
-    (print "usda-token-swap.burn")
-    (print contract-caller)
-    (print amount)
-    ;; TODO - make dynamic
-    ;; (asserts! (is-eq contract-caller .arkadiko-swap-v1-1) (err ERR-NOT-AUTHORIZED))
-    (ft-burn? yield-wbtc-59760-wbtc amount recipient)
+    (asserts! (is-eq contract-caller (var-get contract-owner)) ERR-NOT-AUTHORIZED)
```

Comment:
Same as the other comments, using `contract-caller` creates a strict contract chain.